crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -902,8 +907,39 @@ impl CumulocityConverter {
             _ => {
                 // if device is unregistered register using auto-registration
                 if self.entity_store.get(&source).is_none() {
-                    let auto_registration_messages =
-                        self.entity_store.auto_register_entity(&source)?;
+                    let auto_registration_messages = match self
+                        .entity_store
+                        .auto_register_entity(&source)
+                    {
+                        Ok(auto_registration_messages) => auto_registration_messages,
+                        Err(e) => match e {
+                            entity_store::Error::NonDefaultTopicScheme(eid) => {
+                                // if its non supported enity, then cache it, log
+                                // and also send out error message on `errors` topic only at the first time.
+                                if !self.custom_entities.contains(&eid) {
+                                    if self.custom_entities.len() >= CUSTOM_ENTITY_CACHE_SIZE {
+                                        self.custom_entities.pop_front();
+                                    }
+                                    self.custom_entities.push_back(eid.clone());
+                                    warn!(
+                                    "Auto registration failed for the custom entity due to: {}, try explicitly registering the entity manually",
```

Comment:
The error message of the `NonDefaultTopicScheme` error itself could have been updated with that message, to keep the log statement and the `te/error` messages consistent. The extra advantage is that we can even just `?` that error from here so that it's logged and sent to the `te/errors` topic automatically from the `convert` function higher up the stack.

The updated error message could be something like this:
```rust
    #[error("Auto registration of the entity with topic id {0} failed as it does not match the default topic scheme: 'device/<device-id>/service/<service-id>'. Try explicit registration instead.")]
    NonDefaultTopicScheme(EntityTopicId),
```
I'd drop the word `manually` as it carries connotations of a manual step to be performed.

Reply:
The only side effect of using `?` for the same error is that it will be logged as an error instead of a warning. But I think that's fine since it is only logged once and it genuinely is a user error.

Reply:
Updated the `error` message.